### PR TITLE
Update ruff linter rules according to documentation

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
Enables more ruff linting rules as suggested by [the documentation](https://docs.astral.sh/ruff/linter/#rule-selection). Also disables some that interfere with some core functionality. We can evaluate removing these in the future.
## Issue ticket number and link
NA.

### CODE REVIEW

**Comment by collindutter** [comment]
I'm on the fence about [SIM108](https://docs.astral.sh/ruff/rules/if-else-block-instead-of-if-exp/), but I think i'm leaning towards including it for consistency.

File: griptape/drivers/file_manager/amazon_s3_file_manager_driver.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def try_load_file(self, path: str) -> bytes:
             return response["Body"].read()
         except botocore.exceptions.ClientError as e:
             if e.response["Error"]["Code"] in {"NoSuchKey", "404"}:
-                raise FileNotFoundError
+                raise FileNotFoundError from e
```
Comment (dylanholmes) [inline griptape/drivers/file_manager/amazon_s3_file_manager_driver.py:58]: TIL - I've actually never seen this syntax before, cool!

File: pyproject.toml [hunk 2]
```diff
@@ -200,7 +200,22 @@ line-length = 120
 skip-magic-trailing-comma = true
 
 [tool.ruff.lint]
-select = ["E4", "E7", "E9", "F", "TID251"]
+select = [
+    "E", # pycodestyle
```
Comment (dylanholmes) [inline pyproject.toml:204]: Thanks for adding the comments!

**Review by dylanholmes — APPROVED** [review]
Love the consistency

**Pull request merged**